mart/utils/adapters.py

```diff
@@ -5,34 +5,51 @@
 # agreement between Intel Corporation and you.
 #
 
-__all__ = ["CallableAdapter"]
+from typing import Any, Callable
+
+__all__ = ["CallableAdapter", "PartialAdapter"]
 
 
 class CallableAdapter:
     """Adapter to make an object callable."""
 
-    def __init__(self, instance, redirecting_fn):
+    def __init__(self, redirecting_fn):
         """
 
         Args:
-            instance (object): instance to make callable.
             redirecting_fn (str): name of the function that will be invoked in the `__call__` method.
         """
-        assert instance is not None
         assert redirecting_fn != ""
 
-        self.instance = instance
         self.redirecting_fn = redirecting_fn
 
-    def __call__(self, *args, **kwargs):
+    def __call__(self, instance):
         """
 
         Args:
-            args (Any): values to use in the callable method.
-            kwargs (Any): keyword values to use in the callable method.
+            instance (object): instance to make callable.
         """
-        function = getattr(self.instance, self.redirecting_fn)
 
+        function = getattr(instance, self.redirecting_fn)
         assert callable(function)
+        return function
+
+
+class PartialAdapter:
```

Comment:
I would probably name this `PartialInstanceWrapper` or just `InstanceWrapper`.

Reply:
Good idea. Renamed as `PartialInstanceWrapper`.